### PR TITLE
feat: check if refuce motion enabled, update animation scale

### PR DESCRIPTION
**Pull request by marchenk0va**
### Motivation

Update animation scale to 0, when reduce motion is enabled.

### CODE REVIEW

**Comment by Trancever** [comment]
@marchenk0va The missing piece in this implementation is passing a theme to ThemeProvider. We should check in render if `this.props.theme` evaluates to false, if yes, then we should pass a `defaultTheme` with `animation.scale` set to 0 or 1 based on the `reduceMotionEnabled` state. If theme evaluates to true we just pass it.

Basically this functionality should work only when a user of the lib does not pass custom theme

File: src/core/Provider.tsx [hunk 1]
```diff
@@ -10,8 +12,37 @@ type Props = {
...
+  }
+
+  componentWillUnmount() {
+    AccessibilityInfo.removeEventListener(
+      'reduceMotionEnabled',
+      this._handleReduceMotionToggled
+    );
+  }
+
+  _handleReduceMotionToggled = (reduceMotionEnabled: any) => {
```
Comment (satya164) [inline src/core/Provider.tsx:42]: Also should mark this as `private`

File: src/core/Provider.tsx [hunk 2]
```diff
@@ -1,4 +1,6 @@
+// @ts-nocheck
```
[the remark this answers is not in the document]
Comment (marchenk0va) [reply src/core/Provider.tsx:1]: I removed it. `AccessibilityInfo` is untyped, it throws error on unresolved types for Accessibility methods.

**Comment by Trancever** [comment]
Closing this in favour of #1937

**Pull request closed without merging**